x/authz/keeper/grpc_query.go

```diff
@@ -33,29 +33,35 @@ func (k Keeper) Grants(c context.Context, req *authz.QueryGrantsRequest) (*authz
 	if err != nil {
 		return nil, err
 	}
-	ctx := sdk.UnwrapSDKContext(c)
-
-	store := ctx.KVStore(k.storeKey)
-	key := grantStoreKey(grantee, granter, "")
-	authStore := prefix.NewStore(store, key)
 
+	ctx := sdk.UnwrapSDKContext(c)
 	if req.MsgTypeUrl != "" {
-		authorization, expiration := k.GetCleanAuthorization(ctx, grantee, granter, req.MsgTypeUrl)
-		if authorization == nil {
+		grant, found := k.getGrant(ctx, grantStoreKey(grantee, granter, req.MsgTypeUrl))
+		if !found {
 			return nil, status.Errorf(codes.NotFound, "no authorization found for %s type", req.MsgTypeUrl)
 		}
+
+		authorization := grant.GetAuthorization()
+		if authorization == nil {
+			return nil, status.Errorf(codes.Internal, "expected authz.Authorization")
+		}
```

Comment:
this lines are repeated (in keeper.go), let's use a function and call that function in both places. We can use `errors.ErrNotFound` in both places when we check `!found`